src/app/service/event.service.ts

```diff
@@ -3,58 +3,56 @@ import { HttpClient, HttpParams } from '@angular/common/http';
 import { BehaviorSubject, Observable, catchError, finalize } from 'rxjs';
 import { EventRequest } from '../entity/EventRequest';
 import { UserProfile } from '../entity/UserProfile';
+import { Bet } from '../entity/Bet';
 
 @Injectable({
   providedIn: 'root',
 })
 export class EventService {
-  private apiUrl = 'http://localhost:8080/api/v1/events'; // Adjust the base URL as needed
-  private addUrl = 'http://localhost:8080/api/v1/events/add';
-  private getEventsUrl = 'http://localhost:8080/api/v1/events'; // Adjust the URL as needed
+  private apiUrl = 'http://localhost:8080/api/v1/events';
 
   private loadingSubject = new BehaviorSubject<boolean>(false);
   public loading$: Observable<boolean> = this.loadingSubject.asObservable();
 
   constructor(private http: HttpClient) {}
 
   addEvent(eventRequest: EventRequest): Observable<any> {
-    return this.http.post(this.addUrl, eventRequest);
+    return this.http.post('http://localhost:8080/api/v1/events/add', eventRequest);
+  }
+
+  addBetToEvent(eventId: number, bet: Bet): Observable<any> {
+    return this.http.post(`${this.apiUrl}/bet/${eventId}`, bet);
   }
 
   getEvents(): Observable<EventRequest[]> {
     return this.http.get<EventRequest[]>(this.apiUrl);
   }
 
   getEventById(eventId: string): Observable<EventRequest> {
-    const getUrl = `${this.apiUrl}/get/${eventId}`;
-    return this.http.get<EventRequest>(getUrl);
+    return this.http.get<EventRequest>(`${this.apiUrl}/get/${eventId}`);
   }
 
   updateEvent(eventId: string, eventRequest: EventRequest): Observable<any> {
-    const updateUrl = `${this.apiUrl}/edit/${eventId}`;
-    return this.http.put(updateUrl, eventRequest);
+    return this.http.put(`${this.apiUrl}/edit/${eventId}`, eventRequest);
   }
 
   deleteEvent(eventId: string): Observable<any> {
-    const deleteUrl = `${this.apiUrl}/delete/${eventId}`;
-    return this.http.delete(deleteUrl);
+    return this.http.delete(`${this.apiUrl}/delete/${eventId}`);
   }
 
   getUserProfiles(): Observable<UserProfile[]> {
-    return this.http.get<UserProfile[]>(
-      'http://localhost:8080/api/user-profiles',
-    );
+    return this.http.get<UserProfile[]>('http://localhost:8080/api/v1/user-profile');
   }
```

Comment:
Why is this method needed ? And wouldn't it be better placed in the user profile service ?